### PR TITLE
Add statsD receiver documentation

### PR DESCRIPTION
**Pull request by gavindoudou**
Add statsD receiver documentation.
Test for statsD receiver:
PR to testframework: https://github.com/aws-observability/aws-otel-test-framework/pull/217
PR to aoc: https://github.com/aws-observability/aws-otel-collector/pull/330/files

### CODE REVIEW

**Comment by mxiamxia** [comment]
we'll hold merge on this PR.

File: src/docs/components/statsd-receiver.mdx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    StatsD receiver is an agent that collects [StatsD metrics](https://github.com/statsd/statsd/blob/master/docs/metric_types.md)
+    and do the aggregation in a customer defined interval(default value is 60s). You can send counters and gauges to StatsD receiver.
+    The StatsD receiver will send the aggregated metrics to the following workflow.
+path: '/docs/components/statsd-receiver'
+---
+
+import SectionSeparator from "components/MdxSectionSeparator/sectionSeparator.jsx"
+
+StatsD receiver is an agent that collects [StatsD metrics](https://github.com/statsd/statsd/blob/master/docs/metric_types.md)
+and do the [aggregation](#aggregation) during a customer defined interval (default value is 60s). The aggregation interval is similar to the flush
```
Comment (anuraaga) [inline src/docs/components/statsd-receiver.mdx:13]:
```suggestion
and does [aggregation](#aggregation) for a defined interval (default value is 60s). The aggregation interval is similar to the flush
```

File: src/docs/components/statsd-receiver.mdx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+interval in StatsD. You can send counters and gauges to StatsD receiver. The StatsD receiver will send the aggregated metrics to the following workflow.
```
Comment (anuraaga) [inline src/docs/components/statsd-receiver.mdx:14]:
```suggestion
interval in StatsD. You can send counters and gauges to StatsD receiver, which will send the aggregated metrics to the following workflow.
```

File: src/docs/components/statsd-receiver.mdx [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+For example, you can use EMF exporter to send the metrics to CloudWatch.
+
+StatsD receiver now supports counter and gauge types which covers most of use cases. It listens to a port which can be
```
Comment (anuraaga) [inline src/docs/components/statsd-receiver.mdx:17]:
```suggestion
StatsD receiver currently supports counter and gauge types. The StatsD receiver can be used as a replacement
for the CloudWatch agent if these are the only types you use.
```

File: src/docs/components/statsd-receiver.mdx [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+treated as a replacement of CloudWatch agent for StatsD.
+
+<SectionSeparator />
+
+## Enabling StatsD Receiver
+
+To enable the StatsD receiver, add the name under receiver section in the config file `(local/config.yaml)`.
```
Comment (anuraaga) [inline src/docs/components/statsd-receiver.mdx:24]:
```suggestion
To enable the StatsD receiver, add the name under the receiver section in the config file `(local/config.yaml)`.
```

File: src/docs/components/statsd-receiver.mdx [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+By default, the receiver listens address `localhost:8125`, collects all metrics and aggregate the metrics in 60s.
+The `transport` can only be udp.
+(For the full list of metrics, see [Available Metrics](#available-metrics)).
+
+The following configuration collects StatsD metrics by using StatsD receiver and sends
+them to CloudWatch using `awsemf` exporter. You could change the port name, aggregation_interval et.al to the values
+suits you the most. Check out [SETUP](https://aws-otel.github.io/docs/setup/ecs) section for
```
Comment (anuraaga) [inline src/docs/components/statsd-receiver.mdx:31]:
```suggestion
them to CloudWatch using `awsemf` exporter. You can set these configuration values to what works
for your application. Check out [SETUP](https://aws-otel.github.io/docs/setup/ecs) section for
```

File: src/docs/components/statsd-receiver.mdx [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+`<name>:<value>|c|@<sample-rate>|#<tag1-key>:<tag1-value>`
+
+It supports sample rate.
+
+### Gauge
+
+`name>:<value>|g|@<sample-rate>|#<tag1-key>:<tag1-value>`
+
+It does not support sample rate.
```
Comment (anuraaga) [inline src/docs/components/statsd-receiver.mdx:69]:
```suggestion
Sample rate is not supported for gauges.
```

File: src/docs/components/statsd-receiver.mdx [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+
+## Aggregation
+
+Counters and gauges are aggregated separately.
+The examples below mean receiving metrics in the same interval.
```
Comment (anuraaga) [inline src/docs/components/statsd-receiver.mdx:74]: Not sure what this means

Comment (gavindoudou) [reply src/docs/components/statsd-receiver.mdx:74]: Will revise this part.

**Comment by gavindoudou** [comment]
Revised the PR.

**Review by anuraaga — APPROVED** [review]

**Review by mxiamxia — APPROVED** [review]
Thanks!

**Review by alolita — APPROVED** [review]

**Pull request merged**